### ISSUE TITLE
codemirror-promql: Custom prometheus clients are not being used

### ISSUE THREAD

**Issue by Spikatrix** [prometheus/prometheus#11012]
`codemirror-promql` uses the default prometheus client even after [overriding it](https://github.com/prometheus/prometheus/tree/main/web/ui/module/codemirror-promql#override-the-default-prometheus-client).

I've looked into the source and found that the [`isPrometheusConfig` function](https://github.com/prometheus/prometheus/blob/main/web/ui/module/codemirror-promql/src/complete/index.ts#L34-L45) is responsible for the issue. It doesn't do a good job of differentiating between `PrometheusConfig` and `PrometheusClient` objects. For instance:

```typescript
import { HTTPPrometheusClient } from '@prometheus-io/codemirror-promql/dist/esm/client/prometheus';
console.log(isPrometheusConfig(new HTTPPrometheusClient({})))
// Prints true instead of false
```

Basically, any PrometheusClient which has a field named `url`, `lookbackInterval`, `httpErrorHandler`, `fetchFn`, `cache`, `httpMethod` or `apiPrefix` with a value that isn't `undefined` will make the function return true, effectively ignoring the custom client and treating it as a PrometheusConfig.

**Comment by Nexucis** [comment]
mmm that's annoying indeed. But I'm wondering how we can fix it. Or are we limited by typescript and the way to make the difference between two types ?

**Comment by Nexucis** [comment]
actually I found another way to differentiate the config and the client and created a PR to fix it in #11018. We should be good like that

**Comment by Spikatrix** [comment]
That should do the trick. Thanks for fixing it!
